shanoir-ng-datasets/src/main/java/org/shanoir/ng/importer/service/DicomSRImporterService.java

```diff
@@ -0,0 +1,399 @@
+								if (graphicDataSequence != null) {
+									Attributes graphicDataAttributes = graphicDataSequence.get(0);
+									if (graphicDataAttributes != null) {
+										String graphicData = graphicDataAttributes.getString(Tag.GraphicData);
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Create the necessary dataset expression.
+	 * 
+	 * @param datasetAttributes
+	 * @param measurementDataset
+	 * @throws MalformedURLException
+	 */
+	private void createDatasetExpression(Attributes datasetAttributes, MeasurementDataset measurementDataset)
+			throws MalformedURLException {
+		DatasetExpression expression = new DatasetExpression();
+		expression.setCreationDate(LocalDateTime.now());
+		expression.setDatasetExpressionFormat(DatasetExpressionFormat.DICOM);
+		expression.setDataset(measurementDataset);
+		measurementDataset.setDatasetExpressions(Collections.singletonList(expression));		
+		List<DatasetFile> files = createDatasetFiles(datasetAttributes, expression);
+		expression.setDatasetFiles(files);
+	}
+
+	/**
+	 * Create the dataset files, as WADO-RS links in that case,
+	 * as OHIF viewer works only with new version of dcm4chee (arc-light 5.x).
+	 * 
+	 * @param datasetAttributes
+	 * @param expression
+	 * @return
+	 * @throws MalformedURLException
+	 */
+	private List<DatasetFile> createDatasetFiles(Attributes datasetAttributes, DatasetExpression expression)
+			throws MalformedURLException {
+		DatasetFile datasetFile = new DatasetFile();
+		final String studyInstanceUID = datasetAttributes.getString(Tag.StudyInstanceUID);
+		final String seriesInstanceUID = datasetAttributes.getString(Tag.SeriesInstanceUID);
+		final String sOPInstanceUID = datasetAttributes.getString(Tag.SOPInstanceUID);
+		final StringBuffer wadoStrBuf = new StringBuffer();
+		wadoStrBuf.append(dcm4cheeProtocol + dcm4cheeHost + ":" + dcm4cheePortWeb);
+		wadoStrBuf.append(dicomWebRS + "/" + studyInstanceUID
+					+ "/series/" + seriesInstanceUID + "/instances/" + sOPInstanceUID);
+		URL wadoURL = new URL(wadoStrBuf.toString());
+		datasetFile.setPath(wadoURL.toString());
+		datasetFile.setPacs(true);
+		datasetFile.setDatasetExpression(expression);
+		List<DatasetFile> files = new ArrayList<DatasetFile>();
+		files.add(datasetFile);
+		return files;
+	}
+
+	/**
+	 * This method writes both attributes to an output stream and converts
+	 * this one to an input stream, that can be used to send the manipulated
+	 * file to the backend pacs.
+	 * 
+	 * @param metaInformationAttributes
+	 * @param datasetAttributes
+	 * @throws IOException
+	 * @throws Exception
+	 */
+	private void sendToPacs(Attributes metaInformationAttributes, Attributes datasetAttributes)
+			throws IOException, Exception {
+		/**
+		 * Create a new output stream to write the changes into and use its bytes
+		 * to produce a new input stream to send later by http client to the DICOM server.
+		 */
+		ByteArrayOutputStream bAOS = new ByteArrayOutputStream();
```

Comment:
Use a try with ressources ?

Reply:
done, added, but on the outer stream, that closes the inner as well